.github/workflows/build-deploy-cloudrun-function.yml

```diff
@@ -0,0 +1,35 @@
+name: Build & Deploy Cloudrun Functions
+
+on:
+  push:
+    branches:
+      - main
+      - staging
+    paths:
+      - 'functions/fivetran-trigger/**'
+      - 'functions/function2/**'
+  workflow_dispatch:
+    inputs:
+      function_name:
+        description: 'The name of the Cloud Function to deploy'
+        required: true
+
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set function name
+        id: set-function-name
+        run: |
+          case "${{ github.event.head_commit.message }}" in
+            *"functions/fivetran-trigger"*) echo "::set-output name=function_name::fivetran-trigger" ;;
+            *"functions/function2"*) echo "::set-output name=function_name::function2" ;;
+          esac
```

Comment:
this looks like it'll deploy at most one function. Maybe not ideal if we want to update dependencies on both in the same PR. But, like I said in slack, not worth refining it now.